File: typespec/admin/company/regional.go
// Package company contains Admin API company-setting wire types.
package company

import (
	admincommon "github.com/vetchium/src/typespec/admin/common"
	"github.com/vetchium/src/typespec/common"
)

type CompanyRegionalDefaultsResponse struct {
	DefaultLanguage admincommon.LanguageCode `json:"default_language"`
	DefaultTimezone common.TimeZoneID        `json:"default_timezone"`
}

type SetCompanyRegionalDefaultsRequest struct {
	DefaultLanguage admincommon.LanguageCode `json:"default_language"`
	DefaultTimezone common.TimeZoneID        `json:"default_timezone"`
}

func (r SetCompanyRegionalDefaultsRequest) Validate() []string {
	fields := make([]string, 0, 2)
	if !admincommon.IsLanguageCode(r.DefaultLanguage) {
		fields = append(fields, "default_language")
	}
	if !common.IsTimeZoneID(r.DefaultTimezone) {
		fields = append(fields, "default_timezone")
	}
	return fields
}

File: typespec/admin/company/regional.ts
import {
  isLanguageCode,
  isTimeZoneID,
  type LanguageCode,
  type TimeZoneID,
} from "../../common/localization.ts";

export interface CompanyRegionalDefaultsResponse {
  default_language: LanguageCode;
  default_timezone: TimeZoneID;
}

export interface SetCompanyRegionalDefaultsRequest {
  default_language: LanguageCode;
  default_timezone: TimeZoneID;
}

export function validateSetCompanyRegionalDefaultsRequest(
  request: SetCompanyRegionalDefaultsRequest,
): string[] {
  const fields: string[] = [];
  if (!isLanguageCode(request.default_language)) {
    fields.push("default_language");
  }
  if (!isTimeZoneID(request.default_timezone)) {
    fields.push("default_timezone");
  }
  return fields;
}
